.github/workflows/desktop.yml

```diff
@@ -60,106 +60,37 @@ jobs:
-        if: startsWith(matrix.generator, 'Ninja') && startsWith(matrix.os, 'macos')
-        run: |
-          brew install ninja
-
-      - name: Install Ninja (linux)
-        if: startsWith(matrix.generator, 'Ninja') && startsWith(matrix.os, 'ubuntu')
-        run: |
-          sudo apt install ninja-build
-
-      - name: Install system level applications (mac)
-        if: startsWith(matrix.os, 'macos')
-        id: macos-brew-install
-        run: |
-          brew install ccache
-          echo "::set-env name=CCACHE_INSTALLED::1"
-
-      - name: Install system level applications (linux)
-        if: startsWith(matrix.os, 'ubuntu')
-        id: ubuntu-apt-install
-        run: |
-          sudo apt install ccache
-          echo "::set-env name=CCACHE_INSTALLED::1"
-
       - name: Cache vcpkg C++ dependencies
         id: cache_vcpkg
         uses: actions/cache@v2
         with:
           path: external/vcpkg/installed
           key: dev-vcpkg-${{ matrix.vcpkg_triplet }}-${{ hashFiles(format('{0}', env.VCPKG_RESPONSE_FILE)) }}-${{ hashFiles('.git/modules/external/vcpkg/HEAD') }}
 
-      - name: Install vcpkg packages (windows)
-        if: steps.cache_vcpkg.outputs.cache-hit!='true' && startsWith(matrix.os, 'windows')
-        run: |
-          ${{ github.workspace }}/external/vcpkg/bootstrap-vcpkg.bat -disableMetrics
-          ${{ github.workspace }}/external/vcpkg/vcpkg install @${{env.VCPKG_RESPONSE_FILE}}
-
-      - name: Install vcpkg packages (mac and linux)
-        if: steps.cache_vcpkg.outputs.cache-hit!='true' && startsWith(matrix.os, 'windows')!='true'
-        run: |
-          ${{ github.workspace }}/external/vcpkg/bootstrap-vcpkg.sh -disableMetrics
-          ${{ github.workspace }}/external/vcpkg/vcpkg install @${{env.VCPKG_RESPONSE_FILE}}
+      - name: Cache ccache files
+        if: startsWith(matrix.os, 'ubuntu') || startsWith(matrix.os, 'macos')
+        id: cache_ccache
+        uses: actions/cache@v2
+        with:
+          path: ccache_dir
+          key: dev-test-ccache-${{ env.MATRIX_UNIQUE_NAME }}
 
       - name: Setup python
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python_version }}
           architecture: ${{ matrix.architecture }}
 
-      - name: Get pip cache dir
-        id: get_pip_cache_dir
+      - name: Install prerequisites(cpp and python dependencies)
         run: |
-          echo "::set-output name=cachedir::$(pip cache dir)"
-
-      - name: pip cache
-        id: cache_pip
-        uses: actions/cache@v2
-        with:
-          path: ${{ steps.get_pip_cache_dir.outputs.cachedir }}
-          key: dev-pip-${{ runner.os }}-${{ matrix.python_version }}-${{ matrix.architecture }}-${{ hashFiles(format('{0}', env.PIP_REQUIREMENTS_FILE)) }}
+          python scripts/install_prereqs.py --ccache --generator "${{ matrix.generator}}"
 
-      - name: Install pip packages
+      - name: CMake configure and build
```

Comment:
nit: Just "Build SDK" or similar is fine, no need to specify cmake here since we're just running the build script.

Reply:
Makes sense.